packages/common/src/discussions/edit.ts

```diff
@@ -45,6 +54,16 @@ export async function createDiscussion(
     discussion.typeKeywords,
     discussion.isDiscussable
   );
+  // initialize discussion settings
```

Comment:
We can't create the discussion settings until the item is created, so this needs moved below the call to `createModel`. We also need to update the `requestOptions` arg type to `IHubRequestOptions` then spread it's value onto the object passed to `createSettings` for auth to work correctly to the API.

```jsx
export async function createDiscussion(
  partialDiscussion: Partial<IHubDiscussion>,
  requestOptions: IHubRequestOptions
): Promise<IHubDiscussion> {

// ...

  // create the item
  model = await createModel(model, requestOptions as IUserRequestOptions);
  const defaultSettings = getDefaultEntitySettings('discussion');
  // create the entity settings
  model.entitySettings = await createSetting({
    data: {
      id: model.item.id,
      type: defaultSettings.type,
      settings: {
        ...defaultSettings.settings,
        ...discussion.discussionSettings,
      },
    },
    ...requestOptions,
  });
```